src/openrct2/Context.cpp

```diff
@@ -987,11 +994,13 @@ namespace OpenRCT2
 
         void RunFrame()
         {
+            const auto deltaTime = _timer.GetElapsed();
+            _timer.Restart();
```

Comment:
Should there be a convenience function that does both? Gets the elapsed and then restarts?

Reply:
What would you call it? I was unable to come up with a sensible name that isn't too long and yet describes what it does.

Reply:
What would you call it? I was unable to come up with a sensible name that isn't too long and yet describes what it does.

Reply:
I haven't been able to come up with anything that looks great too, could only think of `getElapsedAndReset()`

Reply:
Why not have `Reset` return the value before the reset? Many timer implementations do that.

Reply:
Not sure if its actually worth the trouble from turning two lines to one by adding a new function to be honest.

Reply:
I think it boils down to whether you'll want to read `GetElapsed` two times in a row without restarting the timer. If this seems to be a good use case, then we should be fine with two functions, but if every time we'll have to read and reset, then a single function seems less error prone.